packages/react/src/components/Combobox/useCombobox.tsx

```diff
@@ -0,0 +1,131 @@
+import React, { useEffect, useState } from 'react';
+
+import type { ComboboxItemProps } from './Item/Item';
+import { ComboboxItem } from './Item/Item';
+
+export type UseComboboxProps = {
+  children: React.ReactNode;
+  input: string;
+  multiple: boolean;
+  activeValues: ValueItemType[];
+  filter: (inputValue: string, label: string, value: string) => boolean;
+};
+
+export type ValueItemType = {
+  value: string;
+  label: string;
+};
+
+export default function useCombobox({
+  children,
+  input,
+  multiple,
+  filter,
+}: UseComboboxProps) {
+  const [open, setOpen] = useState(false);
+  const [values, setValues] = useState<Set<ValueItemType>>(new Set([]));
+
+  // Update all values
+  useEffect(() => {
+    const allValues: ValueItemType[] = [];
+    React.Children.forEach(children, (child) => {
+      if (React.isValidElement(child) && child.type === ComboboxItem) {
+        const props = child.props as ComboboxItemProps;
+
+        let label = '';
+        // get label from children, but only the string
+        React.Children.forEach(props.children, (child) => {
+          if (typeof child === 'string') {
+            label += child;
+          }
+        });
+
+        allValues.push({
+          value: props.value,
+          label,
+        });
+      }
+    });
+    const allValuesSet = new Set(allValues);
+    setValues(allValuesSet);
+  }, [children]);
+
+  const getComboboxItems = () => {
+    const valuesArray = Array.from(values);
+    const childrenArr = React.Children.toArray(children).filter((child) => {
+      if (!React.isValidElement(child)) return false;
+      if (child.type !== ComboboxItem) return false;
+      return true;
+    });
+
+    const activeValue = valuesArray.find((item) => item.label === input);
+    if (activeValue && !multiple) {
+      return childrenArr;
+    }
+
+    if (input === '' && !multiple) return childrenArr;
+
+    return childrenArr.filter((child) => {
+      if (!React.isValidElement(child)) return false;
+      if (child.type !== ComboboxItem) return false;
```

Comment:
This is redundant as `childrenArr` was already filtered using this predicate on line 55.